superset/views.py

```diff
@@ -2026,6 +2026,42 @@ def fave_dashboards(self, user_id):
 
     @api
     @has_access_api
+    @expose("/fave_queries/", methods=['GET'])
+    def fave_queries(self):
+        qry = (
+            db.session.query(
+                models.Query,
+            )
+            .join(
+                models.FavStar,
+                sqla.and_(
+                    models.FavStar.user_id == int(g.user.id),
+                    models.FavStar.class_name == 'query',
+                    models.Query.client_id == models.FavStar.obj_id,
```

Comment:
you should not be specifying `models.Query.client_id == models.FavStar.obj_id` if there is a foreign key

`subqueryload` could work here: http://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html